runtime/v2/builder.go

```diff
@@ -219,3 +232,13 @@ func AppBuilderWithPostTxExec[T transaction.Tx](
 		a.postTxExec = postTxExec
 	}
 }
+
+func AppBuilderWithPreblocker[T transaction.Tx](
```

Comment:
Can we add go docs here?

Reply:
We should precise as well that the module preblocker are ran before any logic there.
Which makes me think, do you think we should always run pre blocker before their logic?
What if we give them a callback to call preblocker themselves if needed.

Like

```go
preblocker func(
		ctx context.Context, txs []T, mmPreblocker func()
	) error,
```

And then above we do smth like

```go
	preblockerFn := func(ctx context.Context, txs []T, mmPreblocker func()) error {
		if a.preblocker != nil {
			return a.preblocker(ctx, txs, func() { if err := a.app.moduleManager.PreBlocker()(ctx, txs); err != nil {
			panic(err)
		} })
		}

                 return .app.moduleManager.PreBlocker()(ctx, txs)
	}
```

Reply:
I like this, adding it